mypy/main.py

```diff
@@ -252,6 +257,89 @@ def infer_python_executable(options: Options,
   Define MYPY_CACHE_DIR to override configuration cache_dir path."""  # type: Final
 
 
+class CapturableArgumentParser(argparse.ArgumentParser):
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        try:
+            self.stdout = kwargs.pop('stdout')
+        except KeyError:
+            self.stdout = sys.stdout
+        try:
+            self.stderr = kwargs.pop('stderr')
+        except KeyError:
+            self.stderr = sys.stderr
+        super().__init__(*args, **kwargs)
+
+    # =====================
+    # Help-printing methods
+    # =====================
+    def print_usage(self, file: Optional[IO[str]] = None) -> None:
+        if file is None:
+            file = self.stdout
+        self._print_message(self.format_usage(), file)
+
+    def print_help(self, file: Optional[IO[str]] = None) -> None:
+        if file is None:
+            file = self.stdout
+        self._print_message(self.format_help(), file)
+
+    def _print_message(self, message: str, file: Optional[IO[str]] = None) -> None:
+        if message:
+            if file is None:
+                file = self.stderr
+            file.write(message)
+
+    # ===============
+    # Exiting methods
+    # ===============
+    def exit(self, status: int = 0, message: Optional[str] = None) -> NoReturn:
+        if message:
+            self._print_message(message, self.stderr)
+        sys.exit(status)
+
+    def error(self, message: str) -> NoReturn:
+        """error(message: string)
+
+        Prints a usage message incorporating the message to stderr and
+        exits.
+
+        If you override this in a subclass, it should not return -- it
+        should either exit or raise an exception.
+        """
+        self.print_usage(self.stderr)
+        args = {'prog': self.prog, 'message': message}
+        self.exit(2, gettext('%(prog)s: error: %(message)s\n') % args)
+
+
+class CapturableVersionAction(argparse.Action):
```

Comment:
Add docstring (maybe refer to the docstring that I requested for `CapturableArgumentParser`).